ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/MetricsManager.scala

```diff
@@ -3,33 +3,71 @@
 
 package com.daml.ledger.api.benchtool.metrics
 
-import akka.actor.CoordinatedShutdown
+import akka.actor.{Cancellable, CoordinatedShutdown}
 import akka.actor.typed.scaladsl.AskPattern._
 import akka.actor.typed.{ActorRef, ActorSystem, Props, SpawnProtocol}
 import akka.util.Timeout
-import com.daml.ledger.api.benchtool.util.MetricReporter
+import com.daml.ledger.api.benchtool.util.ReportFormatter
+import org.slf4j.LoggerFactory
 
 import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
 
-case class MetricsManager[T](collector: ActorRef[MetricsCollector.Message])(implicit
+case class MetricsManager[T](
+    collector: ActorRef[MetricsCollector.Message],
+    logInterval: FiniteDuration,
+    streamName: String,
+)(implicit
     system: ActorSystem[SpawnProtocol.Command]
 ) {
+  def sendNewValue(value: T): Unit =
+    collector ! MetricsCollector.Message.NewValue(value)
+
+  def result[Result](): Future[StreamResult] = {
+    logger.debug(s"Requesting result of stream: $streamName")
+    periodicRequest.cancel()
```

Comment:
We can cancel periodic request for report because at this point we're asking for a final report. This means that either the stream has completed or we're shutting down because of forced shut down.